packages/core/getUserAvailability.ts

```diff
@@ -413,9 +416,54 @@ const _getUserAvailability = async function getUsersWorkingHoursLifeTheUniverseA
     end: dayjs(busy.end),
   }));
 
-  const dateRangesInWhichUserIsAvailable = subtract(dateRanges, formattedBusyTimes);
+  let dateRangesInWhichUserIsAvailable = subtract(dateRanges, formattedBusyTimes);
+
+  let dateRangesInWhichUserIsAvailableWithoutOOO = subtract(oooExcludedDateRanges, formattedBusyTimes);
+
+  if (considerOnlyFirstSlot === true && dateRangesInWhichUserIsAvailableWithoutOOO.length > 0) {
+    // Get the date range of the whole day to get the initial time of the day.
+    const { dateRanges: wholeDayDateRange } = buildDateRanges({
+      dateFrom: dayjs(dateFrom).startOf("date"),
+      dateTo: dayjs(dateTo).endOf("date"),
+      availability,
+      timeZone,
+      travelSchedules: isDefaultSchedule
+        ? user.travelSchedules.map((schedule) => {
+            return {
+              startDate: dayjs(schedule.startDate),
+              endDate: schedule.endDate ? dayjs(schedule.endDate) : undefined,
+              timeZone: schedule.timeZone,
+            };
+          })
+        : [],
+      outOfOffice: datesOutOfOffice,
+    });
+
+    const wholeDayDateRangeExcludingBusyTimes = subtract(wholeDayDateRange, formattedBusyTimes);
+
+    if (wholeDayDateRangeExcludingBusyTimes.length > 0) {
+      const firstSlotDateRange = {
+        start: wholeDayDateRangeExcludingBusyTimes[0].start,
```

Comment:
The first slot doesn't always start when the date range starts, it also depends on the time-slot intervals set

Reply:
@CarinaWolli let me reaffirm what is expected here based on the top comment and the code comment:

> Now the next open slot of this day is 10:00, booking this slot throws the error
 - There should be no "open slot" at 10:00 since we selected 'Only show the first slot of each day as available' 
 - AND/OR you want a different error other than "No available users found"?
 
 > The first slot doesn't always start when the date range starts, it also depends on the time-slot intervals set
 - we need to update `wholeDayDateRangeExcludingBusyTimes` array based on the interval time slot too.
 
I will create a new PR based on your response.

Reply:
@omar391 Also, we need to take offset and all into consideration.

I gave a shot at this and almost got there, but was still having few edge cases, and will need some more time from my end. I am currently busy with some other things, so if you want you can go ahead. 👍